src/pages/index.js

```diff
@@ -20,42 +33,75 @@ function openPreviewModal(cardData) {
   imageModal.open(cardData);
 }
 
+// Confirm Delete Modal
+const deleteCardModal = new PopupWithConfirm("#card-delete-modal");
+deleteCardModal.setEventListeners();
+
+function openDeleteCardModal(cardId, cardElement) {
+  deleteCardModal.setSubmitAction(() => {
+    deleteCardModal.setIsLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove(); // Remove the card from the DOM immediately after successful deletion
+        deleteCardModal.close();
+      })
+      .catch((err) => {
+        console.error(`Error deleting card: ${err}`);
+        alert("Failed to delete the card. Please try again."); // Optional user feedback
+      })
+      .finally(() => deleteCardModal.setIsLoading(false));
+  });
+  deleteCardModal.open();
+}
+
 function createCard(data) {
-  const card = new Card(data, "#card-template", openPreviewModal);
+  const card = new Card(
+    data,
+    "#card-template",
+    openPreviewModal,
+    (cardId, isLiked) => {
+      const apiCall = isLiked ? api.removeLike(cardId) : api.addLike(cardId);
+      apiCall.then((updatedCard) => {
+        card.updateLikes(updatedCard.likes); // Updates likes dynamically in the UI
+      });
+    },
+    openDeleteCardModal // Pass the delete modal function directly
+  );
   return card.getView();
 }
 
 // Initialize Section for rendering cards
 const cardSection = new Section({
   renderer: (item) => {
-    const cardEl = createCard(item);
-    cardSection.addItem(cardEl);
-  },
-  selector: selectors.cardSelection,
-});
-cardSection.renderItems(initialCards);
-
-// Initialize API
-const api = new Api({
-  baseUrl: "https://around-api.en.tripleten-services.com/v1",
-  headers: {
-    authorization: "c56e30dc-2883-4270-a59e-b2f7bae969c6",
-    "Content-Type": "application/json",
+    const cardEl = createCard(item); // Create and return the card element
+    return cardEl;
   },
+  selector: selectors.cardSelection, // The CSS selector for the card container
 });
 
-// Initialize User Info
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  infoSelector: ".profile__description",
-});
+// Fetch and render cards
+api
+  .getCards()
+  .then((cards) => {
+    cardSection.renderItems(cards.reverse());
+  })
+  .catch((err) => console.error(`Error fetching cards: ${err}`));
 
 // Profile Edit Modal
 const profileModal = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: (formData) => {
-    userInfo.setUserInfo(formData);
-    profileModal.close();
+    api
```

Comment:
* According to the task, you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.



`  profileModal.setLoading(true)` is needed here to change the button text